File: kinde_flutter_sdk_android/android/src/main/kotlin/com/example/kinde_flutter_sdk_android/models/KindeClientOptions.kt
package com.example.kinde_flutter_sdk_android.models

import java.io.Serializable

data class KindeClientOptions(
  val audience: String? = null,
  val clientId: String,
  val redirectUri: String,
  val domain: String,
  val isDangerouslyUseLocalStorage: Boolean? = null,
  val logoutUri: String? = null,
  val scopes: List<String>,
  val proxyRedirectUri: String? = null,
  val framework: String? = null,
  val frameworkVersion: String? = null
): Serializable {
  companion object {
    fun fromMap(map: Map<String, Any>): KindeClientOptions {
      return KindeClientOptions(
        audience = map["audience"] as? String,
        clientId = map["client_id"] as String,
        redirectUri = map["redirect_uri"] as? String ?: throw IllegalArgumentException("redirect_uri is required"),
        domain = map["domain"] as? String ?: throw IllegalArgumentException("domain is required"),
        isDangerouslyUseLocalStorage = map["isDangerouslyUseLocalStorage"] as? Boolean,
        logoutUri = map["logout_uri"] as? String,
        scopes = map["scopes"] as List<String>? ?: listOf("openid", "offline", "email", "profile"),
        proxyRedirectUri = map["proxy_redirect_uri"] as? String,
        framework = map["framework"] as? String,
        frameworkVersion = map["framework_version"] as? String
      )
    }
  }
}

File: kinde_flutter_sdk_platform_interface/lib/src/models/kinde_client_options.dart
class KindeClientOptions {
  final String?  audience;
  final String?  client_id;
  final String  redirect_uri;
  final String  domain;
  final bool?  isDangerouslyUseLocalStorage;
  final String?  logout_uri;
  // al Function(ErrorProps)? get onErrorCallback;
  // al Function(KindeUser, Map<String, dynamic>?)? onRedirectCallback;
  List<String> scopes;
  final String? proxy_redirect_uri;
  final String? framework;
  final String? framework_version;

  KindeClientOptions({
   this.  audience,
   this.  client_id,
   required this. redirect_uri,
   required this. domain,
   this.  isDangerouslyUseLocalStorage,
   this.  logout_uri,
   required this. scopes,
   this. proxy_redirect_uri,
   this. framework,
   this. framework_version,
  });

  Map<String, dynamic> toJson() => {
    if(audience != null)
      "audience": audience,
    if(client_id != null)
      "client_id": client_id,
    "redirect_uri": redirect_uri,
    "domain": domain,
    if(isDangerouslyUseLocalStorage != null)
      "isDangerouslyUseLocalStorage": isDangerouslyUseLocalStorage,
    if(logout_uri != null)
      "logout_uri": logout_uri,
    "scopes": scopes,
    if(proxy_redirect_uri != null)
      "proxy_redirect_uri": proxy_redirect_uri,
    if(framework != null)
      "framework": framework,
    if(framework_version != null)
      "framework_version": framework_version,
  };
}
